File: src/main/java/no/nav/bidrag/beregn/bidragsevne/beregning/Bidragsevneberegning.java
package no.nav.bidrag.beregn.bidragsevne.beregning;

import java.math.BigDecimal;
import no.nav.bidrag.beregn.bidragsevne.bo.GrunnlagBeregningPeriodisert;
import no.nav.bidrag.beregn.bidragsevne.bo.ResultatBeregning;

public interface Bidragsevneberegning {
    ResultatBeregning beregn(
        GrunnlagBeregningPeriodisert grunnlagBeregningPeriodisert);

    BigDecimal beregnMinstefradrag(
        GrunnlagBeregningPeriodisert grunnlagBeregningPeriodisert);

    BigDecimal beregnSkattetrinnBelop(
        GrunnlagBeregningPeriodisert grunnlagBeregningPeriodisert);

    static Bidragsevneberegning getInstance(){
        return new BidragsevneberegningImpl();
    }


}

File: src/main/kotlin/no/nav/bidrag/beregn/bidragsevne/beregning/BidragsevneBeregning.kt
package no.nav.bidrag.beregn.bidragsevne.beregning

import no.nav.bidrag.beregn.bidragsevne.bo.GrunnlagBeregning
import no.nav.bidrag.beregn.bidragsevne.bo.ResultatBeregning

fun interface BidragsevneBeregning {
    fun beregn(grunnlag: GrunnlagBeregning): ResultatBeregning

    companion object {
        fun getInstance(): BidragsevneBeregning {
            return BidragsevneBeregningImpl()
        }
    }
}
